Automattic/woocommerce-payments
# Add fee details to order notes for successful payments

Fixes #2522

**The refactor PR #4243 should arrive the `develop` branch before this PR.**

#### Changes proposed in this Pull Request

- Add a new class `WC_Payments_Captured_Event_Note` and new util functions to handle and generate the note for fee details. The logic in this class is basically converted from the equivalent JavaScript functions in file `client/payment-details/timeline/map-events.js`.
- The major point is that I am using the same dataset with PR #4243 to test the output of the new class. This ensures the consistency between PHP and JavaScript results.
- When generating notes (two functions `generate_html_note` and `compose_fee_break_down`), I am trying to use only plain HTML tags and inline styles. Benefits: (1) No need to load the custom CSS file, (2) Format still displays correctly in case WCPay is disabled on sites.









#### Screenshots for two cases:

| The most simple  | The most complicated |
| ------------- | ------------- |
| Only base fee  | 100% discount with fx and international fees  |
| ![simple](https://user-images.githubusercontent.com/10045087/167636282-5d8c9ae0-02d5-4846-9ffa-1e32c82a7634.png) | ![discount](https://user-images.githubusercontent.com/10045087/167636316-0f468eab-b735-4b58-9fd3-ecfc19aa6516.png) |

#### Testing instructions





* Try multiple cases of generating payments: only base fee, with capped fee (SEPA Debit), with international cards, with FX, with discount, etc.
* Ensure that formats and strings in order notes and respective transaction pages (under WP Admin > Payment > Transactions > A specific transaction) are the same.

-------------------


**Post merge**
